pkg/webhook/webhook.go

```diff
@@ -53,17 +55,24 @@ type podMutator struct {
 	azureAuthorityHost string
 	proxyImage         string
 	proxyInitImage     string
+	useNativeSidecar   bool
 }
 
 // NewPodMutator returns a pod mutation handler
-func NewPodMutator(client client.Client, reader client.Reader, audience string, scheme *runtime.Scheme) (admission.Handler, error) {
+func NewPodMutator(client client.Client, reader client.Reader, audience string, scheme *runtime.Scheme, k8sConfig *rest.Config) (admission.Handler, error) {
 	c, err := config.ParseConfig()
 	if err != nil {
 		return nil, err
 	}
 	if audience == "" {
 		audience = DefaultAudience
 	}
+
+	useNativeSidecar, err := util.AssertK8sMinVersion(k8sConfig, 1, 28)
```

Comment:
Looks like `SidecarContainers` went beta in 1.29.  With the 3 version supported skew between API server and kubelet, I think 1.32 is earliest version we can enable this.

```suggestion
	useNativeSidecar, err := util.AssertK8sMinVersion(k8sConfig, 1, 32)
```

Reply:
Good spot, updated